mautic/mautic
# Fix startsWith/endsWith/contains for dynamic content

| Q                                      | A
| -------------------------------------- | ---
| Branch?                                | features for 4.0
| Bug fix?                               | yes
| New feature?                           | no
| Deprecations?                          | no
| BC breaks?                             | no
| Automated tests included?              | yes
| Related user documentation PR URL      | 
| Related developer documentation PR URL | 
| Issue(s) addressed                     | 




#### Description:
I've noticed dynamic content filtering doesn't include startsWith/endsWith/contains filtering. This PR added it.



#### Steps to test this PR:
1. Load up [this PR](https://mautibox.com)
2. Create a dynamic content and add filter, for example, based on Zipcode with "start with" operator and two digit (35 for all 35xxx zip codes) 
![image](https://user-images.githubusercontent.com/49391402/97428088-c8df0a00-1915-11eb-9a9b-3ace6c69510e.png)
3. add the dynamic content to a landing page using the block in the builder
4. See that the dynamic content is not displayed event if the contact zipcode match the filters
5. Repeat same for end with and contains. Shouldn't work too. 
5. This PR fixed it
